mne/coreg.py

```diff
@@ -1223,3 +1227,586 @@ def _scale_xfm(subject_to, xfm_fname, mri_name, subject_from, scale,
+    def _has_eeg_data(self):
+        return (self._has_mri_data and
+                len(self._nearest_transformed_high_res_mri_idx_eeg) > 0)
+
+    @property
+    def _nearest_transformed_high_res_mri_idx_rpa(self):
+        return self._nearest_calc.query(
+            apply_trans(self._head_mri_t, self._hsp.rpa))[1]
+
+    @property
+    def _nearest_transformed_high_res_mri_idx_nasion(self):
+        return self._nearest_calc.query(
+            apply_trans(self._head_mri_t, self._hsp.nasion))[1]
+
+    @property
+    def _nearest_transformed_high_res_mri_idx_lpa(self):
+        return self._nearest_calc.query(
+            apply_trans(self._head_mri_t, self._hsp.lpa))[1]
+
+    @property
+    def _has_lpa_data(self):
+        return (np.any(self._mri.lpa) and np.any(self._hsp.lpa))
+
+    @property
+    def _has_nasion_data(self):
+        return (np.any(self._mri.nasion) and np.any(self._hsp.nasion))
+
+    @property
+    def _has_rpa_data(self):
+        return (np.any(self._mri.rpa) and np.any(self._hsp.rpa))
+
+    @property
+    def _head_mri_t(self):
+        trans = rotation(*self._parameters[:3]).T
+        trans[:3, 3] = -np.dot(trans[:3, :3], self._parameters[3:6])
+        # should be the same as np.linalg.inv(self._mri_head_t)
+        return trans
+
+    def _nearest_calc_default(self):
+        return _DistanceQuery(
+            self._processed_high_res_mri_points * self._parameters[6:9])
+
+    @property
+    def _nearest_transformed_high_res_mri_idx_hsp(self):
+        return self._nearest_calc.query(
+            apply_trans(self._head_mri_t, self._hsp.points))[1]
+
+    @property
+    def _processed_high_res_mri_points(self):
+        return self._get_processed_mri_points('high')
+
+    @property
+    def _processed_low_res_mri_points(self):
+        return self._get_processed_mri_points('low')
+
+    def _get_processed_mri_points(self, res):
+        bem = self._mri.bem_low_res if res == 'low' else self._mri.bem_high_res
+        if self._grow_hair:
+            if len(bem.surf.nn):
+                scaled_hair_dist = (1e-3 * self._grow_hair /
+                                    np.array(self._parameters[6:9]))
+                points = bem.surf.rr.copy()
+                hair = points[:, 2] > points[:, 1]
+                points[hair] += bem.surf.nn[hair] * scaled_hair_dist
+                return points
+            else:
+                raise ValueError("Norms missing from bem, can't grow hair")
+                self._grow_hair = 0
+        else:
+            return bem.surf.rr
+
+    @property
+    def _mri_head_t(self):
+        # rotate and translate hsp
+        trans = rotation(*self._parameters[:3])
+        trans[:3, 3] = np.array(self._parameters[3:6])
+        return trans
+
+    @property
+    def _mri_trans_noscale(self):
+        if self._coord_frame == 'head':
+            t = self._mri_head_t
+        else:
+            t = np.eye(4)
+        return t
+
+    @property
+    def _mri_trans(self):
```

Comment:
This is the transformation to apply to the MRI points, depending on the chosen coordinate frame (head or MRI)

The reason we have head and MRI coord frames is to allow people to think about moving the dig points to the MRI (when viewed in the MRI coord frame) or about stretching/reconfiguring the MRI to conform to the head (when viewed in the head coordinate frame). If this seems too complex, we could get rid of it and always use the MRI coordinate frame.

Really though this difference in views can be implemented at the view level rather than the visualization level. It's reasonable in this class to do everything in MRI coordinates I think.

Reply:
I also leaning towards doing everything in MRI space...

Reply:
Agreed, this class can do everything in MRI space and the viewer/GUI can take care of which coordinate frame to display everything in. It's easier to do the transformation at that level I think as it's really a viz problem